File: source/includes/fts-tutorial/sort/near-tutorial.py
import datetime
import pymongo

# connect to your Atlas cluster
client = pymongo.MongoClient('<connection-string>')

# define pipeline
pipeline = [
  {'$search': {
      'index': 'sort-tutorial',
      'near': {
        'path': 'year', 
        'origin': 2015,
        'pivot': 5
        }
    }},
  {'$limit': 5},
  {'$project': {'title': 1, 'released': 1, 'year': 1 }}
]

# run pipeline
result = client['sample_mflix']['movies'].aggregate(pipeline)

# print results
for i in result:
    print(i)

File: source/includes/fts-tutorial/sort/near-tutorial.js
const { MongoClient } = require("mongodb");

// Replace the uri string with your MongoDB deployments connection string.
const uri =
  "<connection-string>";

const client = new MongoClient(uri);

async function run() {
  try {
    await client.connect();

    // set namespace
    const database = client.db("sample_mflix");
    const coll = database.collection("movies");

    // define pipeline
    const agg = [
      {$search: {
        index: "sort-tutorial",
        near: {
          path: "year",
          origin: 2015,
          pivot: 5
        }}},
      {$limit: 5},
      {$project: {"title": 1, "released": 1, "year": 1}}
  ];
           
    // run pipeline
    const result = await coll.aggregate(agg);

    // print results
    await result.forEach((doc) => console.log(doc));
    
  } finally {
    await client.close();
  }
}
run().catch(console.dir);
